tests/builder/app_settings.rs

```diff
@@ -263,6 +263,41 @@ fn infer_subcommands_pass_conflicting_aliases() {
     assert_eq!(m.subcommand_name(), Some("test"));
 }
 
+#[test]
+fn infer_long_flag_pass_conflicting_aliases() {
+    let m = Command::new("prog")
+        .infer_subcommands(true)
+        .subcommand(Command::new("test").long_flag_aliases(["testa", "t", "testb"]))
+        .try_get_matches_from(vec!["prog", "te"])
+        .unwrap();
+    assert_eq!(m.subcommand_name(), Some("test"));
+}
+
+#[test]
+fn infer_long_flag_pass_conflicting_name() {
+    let m = Command::new("prog")
+        .infer_subcommands(true)
+        .subcommand(Command::new("test").long_flag("testa"))
+        .try_get_matches_from(vec!["prog", "te"])
+        .unwrap();
+    assert_eq!(m.subcommand_name(), Some("test"));
+}
+
+#[test]
+fn infer_subcommands_long_flag_fail_with_args2() {
+    let m = Command::new("prog")
+        .infer_subcommands(true)
+        .arg(Arg::new("some"))
+        .subcommand(Command::new("a").long_flag("test"))
+        .subcommand(Command::new("b").long_flag("temp"))
+        .try_get_matches_from(vec!["prog", "te"]);
+    assert!(m.is_ok(), "{:?}", m.unwrap_err().kind());
+    assert_eq!(
+        m.unwrap().get_one::<String>("some").map(|v| v.as_str()),
```

Comment:
What is this testing?  Its just saying that a command name didn't match so it went into `some`.  This doesn't have to deal with long flags.

Reply:
Part of the brain fart garbage. Should be sensible now.